src/runtimes/node/src_files.js

```diff
@@ -1,6 +1,44 @@
+const { promisify } = require('util')
+
+const glob = require('glob')
+
+const pGlob = promisify(glob)
+
 const { getDependencyNamesAndPathsForDependencies, listFilesUsingLegacyBundler } = require('../../node_dependencies')
 const { JS_BUNDLER_ZISI } = require('../../utils/consts')
 
+const getPathsOfIncludedFiles = async (includedFiles, basePath) => {
+  // Some of the globs in `includedFiles` might be exclusion patterns, which
+  // means paths that should NOT be included in the bundle. We need to treat
+  // these differently, so we iterate on the array and put those paths in a
+  // `exclude` array and the rest of the paths in an `include` array.
+  const { include, exclude } = includedFiles.reduce(
+    (acc, path) => {
+      if (path.startsWith('!')) {
+        return {
+          ...acc,
+          exclude: [...acc.exclude, path.slice(1)],
+        }
+      }
+
+      return {
+        ...acc,
+        include: [...acc.include, path],
+      }
+    },
+    { include: [], exclude: [] },
+  )
+  const pathGroups = await Promise.all(
+    include.map((expression) => pGlob(expression, { absolute: true, cwd: basePath, ignore: exclude })),
+  )
+
+  // `pathGroups` is an array containing the paths for each expression in the
+  // `include` array. We flatten it into a single dimension.
+  const paths = pathGroups.reduce((acc, group) => [...acc, ...group], [])
```

Comment:
I think we're using the flat polyfill?

```suggestion
  const paths = pathGroups.flat()
```

Reply:
Ah, perfect! That's much better, thanks.